### PR TITLE
Clear attachments store on finishing creation

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.org/wp/42933

### CODE REVIEW

File: frontend/src/app/core/state/resource-collection.service.ts [hunk 1]
```diff
@@ -96,6 +97,21 @@ export abstract class ResourceCollectionService<T> {
...
 
+  /**
+   * Clear a collection key
+   * @param key Collection key to clear
+   */
+  clear(key:string):void {
+    this
+      .store
+      .update(
+        (store) => ({
```
Comment (Kharonus) [inline frontend/src/app/core/state/resource-collection.service.ts]: can we use here deconstruction, as we are not interested in altering anything in store beside `collections`.

```
({ collections }) => (
  {
    collections: omit(collections, key),
  }
)
```

Comment (Kharonus) [reply frontend/src/app/core/state/resource-collection.service.ts]: maybe we can even drop the lodash dependency and have a private `omit` function ourselves?

**Review by HDinger — APPROVED** [review]
Fine from my side 👍

File: frontend/src/app/core/state/resource-collection.service.ts (hunk 1 above)
Comment (Kharonus) [reply frontend/src/app/core/state/resource-collection.service.ts]: like 

```
function omit(obj: Record<string, unknown>, key: string) {
    // eslint-disable-next-line no-unused-vars
    const {[key]: foo, ...result} = obj;
    return result;
}
```

Comment (oliverguenther) [reply frontend/src/app/core/state/resource-collection.service.ts]: Why should we drop lodash? It's built exactly for this purpose and we use it throughout the application. We want to slowly migrate to lodash-es though for tree-shakability, but other dependencies forbid this the past times I've tried.

Comment (Kharonus) [reply frontend/src/app/core/state/resource-collection.service.ts]: yes, lodash is used widely through the application. Not sure, when the decision was made. In the past, I had quite some trouble with npm libs that were used simply for convenience, as you cannot control what those libs on their end take as dependencies. This is the whole basement for the `low dependency` approach.

I'd take the same argumentation much further, if it was a new dependency, but as lodash is accepted for usage and if we do not plan to eventually drop it completely, we should use it here, as you did.

**Review by Kharonus — APPROVED** [review]

**Pull request merged**